### ISSUE TITLE
Intercept modifies the natural execution result

### ISSUE THREAD

**Issue by ditiem** [mobxjs/mobx#896]
I am exploring how to do a deep intersect on nested stores. I found this strange behaviour and I think it is not intentionally.

```javascript
const {observable, computed, intercept} = mobx;

class Todo {
    @observable title;
    @observable author ;

  constructor(title, author)
  {
        this.title  = title;
        this.author = author ;
  }
}


class Author
{
  @observable name = '' ;
  @observable street = null ;

  constructor ( name )
  {
    this.name = name ;
  }
}


class Street
{
  @observable name = '' ;

  constructor ( name )
  {
    this.name = name ;
  }
}

var todo = new Todo( 'say hi', new Author( 'Peter' ) ) ;

// IF you comment out this line, the program works. Else you will get an exception: 
// Uncaught TypeError: Cannot set property 'name' of null
intercept( todo.author, function ( ) { console.log( '--intercept todo 2 ') } ) ;

console.log( 'modify todo.author.street' ) ;
todo.author.street = new Street( 'long street name' ) ;

console.log( 'modify todo.author.street.name', todo.author.street ) ;
todo.author.street.name = 'around the corner' ;
```

Output:

modify todo.author.street
embedded:51 --intercept todo 2 
embedded:65 modify todo.author.street.name null
embedded:65 Uncaught TypeError: Cannot set property 'name' of null

I did a fiddle:
https://jsfiddle.net/ditiem/cujrjztn/5/

**Comment by urugator** [comment]
You have to return change event from interceptor, otherwise the change is ignored, value (new Street) is not set, and therefore the null pointer error is thrown:
```javascript
intercept(todo.author, function (change) { console.log( '--intercept todo 2 '); return change; } ) ;
```
In you're fiddle there is a different problem - you're trying to intercept `todo.title`, which is a string. String can't be intercepted, only observable object or observable property.

**Comment by ditiem** [comment]
Thank you a lot @urugator. You are right. I checked the documentation and it is written. I started using "observe" and then changed to "intercept", without reading the docs of intercept. My fault.

Thanks for pointing out the todo.title. I was testing/learning how mobx works and forgot to remove that intercept line. The intercept functions is a bit rare to me. I know title is an string, but the setter function of the "todo" will be called when writing todo.title = "new value" ( Example: https://jsfiddle.net/ditiem/f6zs3do5/  ). It is a different way of thinking different from what I am used to.

Thank again for your response.
